File: SmartShop-frontend/src/services/paymentService.js
import api from './api';

export const paymentService = {
    // Record a new payment (Admin only)
    recordPayment: async (data) => {
        const response = await api.post('/payments', data);
        return response.data;
    },

    // Get payments for a specific order
    getPaymentsByOrder: async (orderId) => {
        const response = await api.get(`/payments/order/${orderId}`);
        return response.data;
    },

    // Alias for compatibility
    getByOrderId: async (orderId) => {
        const response = await api.get(`/payments/order/${orderId}`);
        return response.data;
    }
};

export default paymentService;

File: SmartShop/src/main/java/org/SmartShop/service/PaymentService.java
package org.SmartShop.service;

import org.SmartShop.dto.payment.PaymentRequestDTO;
import org.SmartShop.dto.payment.PaymentResponseDTO;
import java.util.List;

public interface PaymentService {
    PaymentResponseDTO recordPayment(PaymentRequestDTO dto);
    List<PaymentResponseDTO> getPaymentsByOrder(Long orderId);
}